integration/test_timescaledb.py

```diff
@@ -0,0 +1,310 @@
+        assert isinstance(row[5], int)  # int_col (INTEGER)
+        assert isinstance(row[6], datetime)  # timestamp_col (TIMESTAMPTZ)
+        assert isinstance(row[7], str)  # nchar_col (CHAR)
+        assert isinstance(row[8], str)  # varchar_col (VARCHAR)
+
+        # Verify specific values
+        assert row[5] == i  # int_col matches index
+        assert row[2] == (i % 2 == 0)  # bool_col alternates
+        assert f"varchar_test_{i}" in row[8]  # varchar contains expected pattern
+
+    # Test precision-specific functionality within the same test
+    cursor = connection.cursor()
+    if timestamp_precision == "milliseconds":
+        # Test millisecond precision
+        test_time = "2019-09-18 01:55:10.123+00"
+        cursor.execute(
+            f"""
+            INSERT INTO {table_name} (time, binary_col, bool_col, double_col, float_col,
+                                    int_col, timestamp_col, nchar_col, varchar_col)
+            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)
+        """,
+            (test_time, b"test", True, 1.23, 4.56, 999, test_time, "MS", "millisecond_test"),
+        )
+
+        cursor.execute(f"SELECT EXTRACT(MILLISECONDS FROM time) FROM {table_name} WHERE int_col = 999")
+        result = cursor.fetchone()
+        assert result[0] >= 10000  # At least 10 seconds worth of milliseconds
+
+    elif timestamp_precision == "microseconds":
+        # Test microsecond precision
+        test_time = "2019-09-18 01:55:10.123456+00"
+        cursor.execute(
+            f"""
+            INSERT INTO {table_name} (time, binary_col, bool_col, double_col, float_col,
+                                    int_col, timestamp_col, nchar_col, varchar_col)
+            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)
+        """,
+            (test_time, b"test", True, 1.23, 4.56, 999, test_time, "US", "microsecond_test"),
+        )
+
+        cursor.execute(f"SELECT EXTRACT(MICROSECONDS FROM time) FROM {table_name} WHERE int_col = 999")
+        result = cursor.fetchone()
+        assert result[0] >= 10000000  # At least 10 seconds worth of microseconds
+
+    cursor.close()
+
+
+@pytest.mark.asyncio
+async def test_timescaledb_async_emit(timescaledb):
+    """Test async emission to TimescaleDB"""
+    (connection, host, port, db_name, user, password, table_name, dsn_url, timestamp_precision, columns_config) = (
+        timescaledb
+    )
+
+    time_format = "%d/%m/%y %H:%M:%S UTC%z"
+    if timestamp_precision in ["milliseconds", "microseconds"]:
+        time_format = "%d/%m/%y %H:%M:%S.%f UTC%z"
+
+    controller = build_flow(
+        [
+            AsyncEmitSource(),
+            TimescaleDBTarget(
+                dsn=dsn_url,
+                table=table_name,
+                time_col="time",
+                columns=columns_config,
+                time_format=time_format,
+                max_events=10,
+            ),
+        ]
+    ).run()
+
+    # Test async emission
```

Comment:
Noisy comment repeats test name and is in the wrong place...
```suggestion
```

Reply:
removed